modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import React, {useEffect, useState} from 'react';
+
+import BarChart from '../../../common/components/bar-chart';
+import ClayIconProvider from '../../../common/context/ClayIconProvider';
+import {getPoliciesChartExpiringPolicies} from '../../../common/services/Policy';
+import {getProductsNames} from '../../../common/services/Products';
+import {
+	currentDateString,
+	nextMonthDate,
+} from '../../../common/utils/dateFormatter';
+
+export default function () {
+	const [policies, setPolicies] = useState([]);
+	const [expirationPoliciesList, setExpirationPoliciesList] = useState({});
+
+	useEffect(() => {
+		getProductsNames().then((response) => {
+			const productList = response?.data?.items;
+			productList.map((productName) => {
+				productName.productValue = 0;
+			});
+
+			const productsListFormatted = {};
+			productList.forEach((teste) => {
+				if (!productsListFormatted[teste?.name]) {
+					productsListFormatted[teste?.name] = 0;
+				}
+				setExpirationPoliciesList({...productsListFormatted});
+			});
+		});
+	}, []);
+
+	useEffect(() => {
+		getPoliciesChartExpiringPolicies(
+			currentDateString[0],
+			currentDateString[1],
+			'01',
+			nextMonthDate[0],
+			nextMonthDate[1],
+			nextMonthDate[2]
+		).then((results) => {
+			const totalFilteredPolicies = results?.data;
+
+			const policiesList = [];
+
+			const policyExpirationDays = 15;
+
+			totalFilteredPolicies?.items.map((policy) => {
+				const policyEndDate = Date.parse(policy?.endDate);
+
+				const currentDate = new Date();
+
+				const differenceOfDays = policyEndDate - currentDate;
+
+				const dayInMilliseconds = 1000 * 60 * 60 * 24;
+
+				const renewalDue =
+					Math.floor(differenceOfDays / dayInMilliseconds) + 1;
+
+				if (renewalDue < policyExpirationDays) {
+					policiesList.push(policy);
+				}
+			});
+
+			setPolicies(policiesList);
+
+			return policiesList;
+		});
+	}, []);
+
+	policies?.map((policy) => {
+		if (!expirationPoliciesList[policy?.productName]) {
+			expirationPoliciesList[policy?.productName] = policy?.termPremium;
+		}
+		else {
+			expirationPoliciesList[policy?.productName] += policy?.termPremium;
+		}
+	});
```

Comment:
Please, change it. You are using a ```.map``` here. This should be used when you need to create a new array.

```suggestion
	policies?.forEach((policy) => {
		if (!expirationPoliciesList[policy?.productName]) {
			return expirationPoliciesList[policy?.productName] = policy?.termPremium;
		}
		
		return expirationPoliciesList[policy?.productName] += policy?.termPremium;
		
	});
```

Reply:
Done.